src/ol/source/CompositeTile.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * @module ol/source/CompositeTile
+ */
+
+import DataTile from '../DataTile.js';
+import DataTileSource from '../source/DataTile.js';
+import EventType from '../events/EventType.js';
+import ReprojDataTile from '../reproj/DataTile.js';
+import TileState from '../TileState.js';
+import {listen, unlistenByKey} from '../events.js';
+
+/**
+ * @typedef {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement|ImageBitmap} ImageLike
+ */
+
+/**
+ * @typedef {Uint8Array|Uint8ClampedArray|Float32Array|DataView} ArrayLike
+ */
+
+/**
+ * Data that can be used with a DataTile.
+ * @typedef {ArrayLike|ImageLike} Data
+ */
+
+/**
+ * @typedef {import("./DataTile.js").default} SourceType
+ */
+
+/**
+ * @typedef {Object} Options
+ * @property {number} [transition] Transition time when fading in new tiles (in milliseconds).
+ * @property {string} [key] Key for use in caching tiles.
+ * @property {Array<SourceType>} sources sources.
+ */
+
+/**
+ * @classdesc
+ * A composite tile source
+ * @api
+ */
+class CompositeTileSource extends DataTileSource {
+  /**
+   * @param {Options} options Source options.
+   */
+  constructor(options) {
+    const sources = options.sources || [];
+
+    const wrapX = sources.every((source) => source.getWrapX());
+    const attributions = (
+      /** @type {import('../Map.js').FrameState} */ frameState,
+    ) =>
+      sources
+        .map((source) => source.getAttributions())
+        .filter(Boolean)
+        .map((func) => func(frameState))
+        .join(' / ');
```

Comment:
Avoid hardcoding a separator.  If you push each attribution to an array the css will format it according to any application specific css.

e.g. custom separator

```
.ol-attribution li:not(:last-child):after {
  content: " / ";
}
```

or separate lines

```
.ol-attribution li {
  display: block;
}
```

The code in ol/source/Raster may be useful.

Reply:
Ah! Thank you!
I'll change it to:
```patch
-        .map((func) => func(frameState))
-        .join(' / ');
+        .flatMap((func) => func(frameState));
```
looks reasonable?